src/rust/ide/lib/analytics/src/remote_log.rs

```diff
@@ -9,27 +9,44 @@ pub use wasm_bindgen::prelude::*;
 
 
 #[wasm_bindgen(inline_js="
-export function _remote_log(msg, data) {
-    if (window !== undefined && window.enso !== undefined && window.enso.remote_log !== undefined) {
-        window.enso.remote_log(msg,{data:data})
+export function _remote_log(msg, value) {
+    if (window !== undefined && window.enso !== undefined && window.enso.remoteLog !== undefined) {
+        try {
+            if (value === undefined && value !== null) {
+                window.enso.remoteLog(msg)
+            } else {
+                window.enso.remoteLog(msg,value)
+            }
+        } catch (error) {
+            console.error(\"Error while logging message. \" + error );
+        }
     } else {
         console.warn(\"Failed to send log message.\")
     }
 }
+
+export function _remote_log_value(msg, field_name, value) {
+    const data = {}
+    data[field_name] = value
+    _remote_log(msg,data)
+}
 ")]
 extern "C" {
-     #[allow(unsafe_code)]
-     fn _remote_log(msg:JsValue,data:JsValue);
+    #[allow(unsafe_code)]
+    fn _remote_log_value(msg:JsValue,field_name:JsValue,value:JsValue);
+    #[allow(unsafe_code)]
+    fn _remote_log(msg:JsValue,value:JsValue);
 }
 
 /// Send the provided public event to our logging service.
-pub fn remote_log<T:Loggable>(message: AnonymousData<T>) {
-    _remote_log(message.0.get(), JsValue::NULL);
+pub fn remote_log_event(message:&str) {
+    _remote_log(JsValue::from(message.to_string()),JsValue::UNDEFINED);
```

Comment:
Why do you pass `JsValue::UNDEFINED` here? Just don't pass it. In JS it will automatically be undefined there!

Reply:
Because the Rust wrapper requires it. We could probably also add a new wrapper method that does not need it, but since none of that is exposed outside this file it doesn't really matter, I think. This means we have the more flexible method and avoid having another JS wrapper.